pulser-core/pulser/backend/results.py

```diff
@@ -171,10 +172,28 @@ def _from_abstract_repr(cls, dict: dict) -> Results:
         return results
 
     def to_abstract_repr(self) -> str:
-        return json.dumps(self._to_abstract_repr(), cls=AbstractReprEncoder)
+        """
+        Serializes this object into a json string.
+        Numpy arrays and torch Tensors are converted into lists,
+        and their original class is lost forever.
+
+        Returns:
+            The json string
+        """
+        abstr_str = json.dumps(
+            self._to_abstract_repr(), cls=AbstractReprEncoder
+        )
+        validate_abstract_repr(abstr_str, "results")
```

Comment:
For this validation, can you make it optional? Like Stefano did in EmulationConfig with `skip_validation`